docs/root/intro/arch_overview/http/http_proxy.rst

```diff
@@ -40,15 +39,17 @@ For further configuration information see the :ref:`HTTP filter configuration do
 Memory usage details
 --------------------
 
-.. attention::
-
-  HTTP dynamic forward proxy support currently can use unbounded memory and is not ready for
-  production use. Circuit breakers and other limits will be added soon.
-
 Memory usage detail's for Envoy's dynamic forward proxy support are as follows:
 
 * Each resolved host/port pair uses a fixed amount of memory global to the server and shared
   amongst all workers.
 * Address changes are performed inline using read/write locks and require no host reallocations.
 * Hosts removed via TTL are purged once all active connections stop referring to them and all used
   memory is regained.
+* The :ref:`max_hosts
+  <envoy_api_field_config.common.dynamic_forward_proxy.v2alpha.DnsCacheConfig.max_hosts>` field can
+  be used to limit the number of hosts that the DNS cache will store at any given time.
```

Comment:
So reading this, it's unclear to me what the relationship is between DNS cache entries and dynamic hosts.  

I'd assume that if one had long lived connections, the DNS entry could be fifod out and one could slowly increase the number of upstream hosts in the forward proxy group, so still have unbounded memory (if a slower attack model).  It looks from above that even if we had a resolved host, if the DNS entry had been flushed we wouldn't reuse that connection, so the number of hosts gets soft-bounded (a really long lived connection per host can still create endless host memory, but as long as there are timeouts, the host would eventually get reclaimed).

Does that sound right, and if so maybe worth clarifying?

Reply:
I think the case you mention is covered already by upstream circuit breakers. I will clarify.